File: BASIC_OF_ALGORITHM/How_to_Solve_N_&_M/NM_2.py
A = [0] * 10 #오름차순에서는 V가 없어도 됩니다..
V = [False] * 10
def go(index, start, n, m):
    if index == m: #index 0 ~ m-1이 다차있다 라는 말입니다.
        for i in range(m):
            print(A[i], end = " ")
        print()
        return
    for i in range(start, n + 1):
       # if V[i]:
        #    continue
      #  V[i] = True
        A[index] = i
        go(index + 1, i + 1, n, m)
       # V[i] = False
    

N, M = input().split()

go(0, 1, int(N), int(M))

File: BASIC_OF_ALGORITHM/How_to_Solve_N_&_M/NM_2.cpp
//중복 불가 오름차순의 수열
#include <iostream>
using namespace std;
//NM2
int A[10];
bool V[10];
void go(int index, int start, int n, int m) {
	if (index == m) {
		for (int i = 0; i < m; i++) {
			cout << A[i] << " ";
		}
		cout << endl;
		return;
	}
	for (int i = start; i <= n; i++) {
		//if (V[i]) continue;
		//V[i] = true;
		A[index] = i;
		go(index + 1, i + 1, n, m);
		//V[i] = false;
	}
	
}

int main() {
	int N, M;
	cin >> N >> M;
	go(0, 1, N, M);






	return 0;
}